test/unit_tests/braket/experimental/autoqasm/test_api.py

```diff
@@ -395,48 +429,60 @@ def qasm_simple_condition(do_cnot: bool) -> bool:
     return do_cnot
 
 
-@aq.main
-def qasm_simple_condition_wrapper(do_cnot: bool):
-    qasm_simple_condition(do_cnot)
+@pytest.fixture
+def qasm_simple_condition_wrapper():
+    @aq.main
+    def qasm_simple_condition_wrapper(do_cnot: bool):
+        qasm_simple_condition(do_cnot)
+
+    return qasm_simple_condition_wrapper
 
 
 @pytest.mark.parametrize("do_cnot", [True, False])
-def test_qasm_simple_condition(do_cnot: bool) -> None:
-    expected = """OPENQASM 3.0;
-def qasm_simple_condition(bool do_cnot) -> bool {
+def test_qasm_simple_condition(do_cnot: bool, qasm_simple_condition_wrapper) -> None:
+    expected = f"""OPENQASM 3.0;
+def qasm_simple_condition(bool do_cnot) -> bool {{
     h __qubits__[0];
-    if (do_cnot) {
+    if (do_cnot) {{
         cnot __qubits__[0], __qubits__[1];
-    }
+    }}
     return do_cnot;
-}
+}}
+bool do_cnot = {"true" if do_cnot else "false"};
 qubit[2] __qubits__;
 bool __bool_0__;
 """
-    expected += f"__bool_0__ = qasm_simple_condition({'true' if do_cnot else 'false'});"
-    assert qasm_simple_condition_wrapper(do_cnot).to_ir() == expected
+    expected += "__bool_0__ = qasm_simple_condition(do_cnot);"
```

Comment:
you could now include this in the string literal above

Reply:
True. However, I'll be updating these tests to use a closure around `do_cnot` instead of defining it as an input, as per [Lauren's comment](https://github.com/amazon-braket/amazon-braket-sdk-python/pull/804/files/073a11bec26ffa65632fdb57f9031b32f0f1cf14#r1398044784), so the expected IR will go back to as before